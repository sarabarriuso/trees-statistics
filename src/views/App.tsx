import { useEffect } from 'react';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';

import '../styles/Main.scss';
import { loadTreesData } from '../models/ecologiServiceModel';
import DashboardView from './DashboardView';

function App(): JSX.Element {
  const DEFAULT_CLASSNAME = 'App';

  useEffect(() => {
    loadTreesData();
  }, []);

  return (
    <div className={DEFAULT_CLASSNAME}>
      <Router>
        <Switch>
          <Route path='/' exact>
            <DashboardView />
          </Route>
          <Route path='/dashboard'>
            <DashboardView />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default App;
