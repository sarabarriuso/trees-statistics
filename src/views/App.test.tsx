import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

export default describe('App Tests', () => {
  it('Renders App successfully', () => {
    // Arrange
    // Act
    const tree = shallow(<App />);

    // Assert
    expect(toJson(tree)).toMatchSnapshot();
  });
});
