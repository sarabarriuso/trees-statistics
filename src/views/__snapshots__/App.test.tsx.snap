// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests Renders App successfully 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <DashboardView />
      </Route>
      <Route
        path="/dashboard"
      >
        <DashboardView />
      </Route>
    </Switch>
  </BrowserRouter>
</div>
`;
